pontoon/terminology/models.py

```diff
@@ -0,0 +1,251 @@
+from __future__ import unicode_literals
+
+from collections import defaultdict
+import logging
+
+from bulk_update.helper import bulk_update
+from django.db import models
+from django.db import transaction
+
+from pontoon.base.models import (
+    Entity,
+    Locale,
+)
+
+from pontoon.base.utils import (
+    get_words,
+    get_singulars
+)
+from pontoon.terminology.term_indexes import DBTermIndex
+
+log = logging.getLogger(__name__)
+
+
+class TermManager(models.Manager):
```

Comment:
I think this is a bad use of the ``Manager`` class. As far as I understand it, the ``Manager`` is made for constructing queries on a database model. Basically, all base methods of the ``models.Manager`` class return a new instance of itself with a different context (with a few exceptions that do not matter, like ``.count()``). So, in my opinion, if your method doesn't return an instance of its class, then it shouldn't be in a Manager. 

However, if I'm wrong regarding this, please correct me!